optimize-data-indexing/OptimizeDataIndexing/Program.cs

```diff
@@ -0,0 +1,180 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Runtime.Serialization.Formatters.Binary;
+using System.Threading;
+using System.Threading.Tasks;
+using Microsoft.Azure.Search;
+using Microsoft.Azure.Search.Models;
+using Microsoft.Extensions.Configuration;
+using Newtonsoft.Json;
+using Index = Microsoft.Azure.Search.Models.Index;
+
+namespace OptimizeDataIndexing
+{
+    class Program
+    {
+        public static async Task Main(string[] args)
+        {
+            IConfigurationBuilder builder = new ConfigurationBuilder().AddJsonFile("appsettings.json");
+            IConfigurationRoot configuration = builder.Build();
+
+            SearchServiceClient serviceClient = CreateSearchServiceClient(configuration);
+
+            string indexName = configuration["SearchIndexName"];
+            ISearchIndexClient indexClient = serviceClient.Indexes.GetClient(indexName);
+
+            Console.WriteLine("{0}", "Deleting index...\n");
+            await DeleteIndexIfExists(indexName, serviceClient);
+
+            Console.WriteLine("{0}", "Creating index...\n");
+            await CreateIndex(indexName, serviceClient);
+
+            Console.WriteLine("{0}", "Finding optimal batch size...\n");
+            await TestBatchSizes(indexClient, numTries: 3);
+
+            //long numDocuments = 100000;
+            //DataGenerator dg = new DataGenerator();
+            //List<Hotel> hotels = dg.GetHotels(numDocuments);
+
+            //Console.WriteLine("{0}", "Uploading using exponential backoff...\n");
```

Comment:
"uploading using exponential back off" ???  If this is creating friction in the push operation, then I think I understand backoff retry correctly.  If that's not the case, and then my suggestions for the readme are not valid.